filament/include/filament/ToneMapper.h

```diff
@@ -0,0 +1,230 @@
+     * function applied ("linear Rec.2020").
+     *
+     * @param c Input color to tone map, in the Rec.2020 color space with no
+     *          transfer function applied ("linear")
+     *
+     * @return A tone mapped color in the Rec.2020 color space, with no transfer
+     *         function applied ("linear")
+     */
+    virtual math::float3 operator()(math::float3 c) const noexcept = 0;
+};
+
+/**
+ * Linear tone mapping operator that returns the input color but clamped to
+ * the 0..1 range. This operator is mostly useful for debugging.
+ */
+struct UTILS_PUBLIC LinearToneMapper final : public ToneMapper {
+    LinearToneMapper() noexcept;
+    ~LinearToneMapper() noexcept final;
+
+    math::float3 operator()(math::float3 c) const noexcept;
+};
+
+/**
+ * ACES tone mapping operator. This operator is an implementation of the
+ * ACES Reference Rendering Transform (RRT) combined with the Output Device
+ * Transform (ODT) for sRGB monitors (dim surround, 100 nits).
+ */
+struct UTILS_PUBLIC ACESToneMapper final : public ToneMapper {
+    ACESToneMapper() noexcept;
+    ~ACESToneMapper() noexcept final;
+
+    math::float3 operator()(math::float3 c) const noexcept;
+};
+
+/**
+ * ACES tone mapping operator, modified to match the perceived brightness
+ * of FilmicToneMapper. This operator is the same as ACESToneMapper but
+ * applies a brightness multiplier of ~1.6 to the input color value to
+ * target brighter viewing environments.
+ */
+struct UTILS_PUBLIC ACESLegacyToneMapper final : public ToneMapper {
+    ACESLegacyToneMapper() noexcept;
+    ~ACESLegacyToneMapper() noexcept final;
+
+    math::float3 operator()(math::float3 c) const noexcept;
+};
+
+/**
+ * "Filmic" tone mapping operator. This tone mapper was designed to
+ * approximate the aesthetics of the ACES RRT + ODT for Rec.709
+ * and historically Filament's default tone mapping operator. It exists
+ * only for backward compatibility purposes and is not otherwise recommended.
+ */
+struct UTILS_PUBLIC FilmicToneMapper final : public ToneMapper {
+    FilmicToneMapper() noexcept;
+    ~FilmicToneMapper() noexcept final;
+
+    math::float3 operator()(math::float3 x) const noexcept;
+};
+
+/**
+ * Generic tone mapping operator that gives control over the tone mapping
+ * curve. This operator can be used to control the aesthetics of the final
+ * image. This operator also allows to control the dynamic range of the
+ * scene referred values.
+ *
+ * The tone mapping curve is defined by 5 parameters:
+ * - contrast: controls the contrast of the curve
+ * - shoulder: controls the shoulder of the curve, i.e. how quickly scene
+ *             referred values map to output white
+ * - midGrayIn: sets the input middle gray
+ * - midGrayOut: sets the output middle gray
+ * - hdrMax: defines the maximum input value that will be mapped to
+ *           output white
+ */
+struct UTILS_PUBLIC GenericToneMapper final : public ToneMapper {
```

Comment:
Because `GenericToneMapper` has an allocated field (`mOptions`) it's not safe to copy it. I would suggest to at least add a deleted copy ctor and operator= and implement the move versions.
```
GenericToneMapper(GenericToneMapper const&) = delete;
GenericToneMapper& operator=(GenericToneMapper const&) = delete;
GenericToneMapper(GenericToneMapper&& rhs)  noexcept : mOptions(rhs.mOptions) { rhs.mOptions = nullptr; }
GenericToneMapper& operator=(GenericToneMapper& rhs) noexcept { /* same as above */ }
```

Reply:
Done.